### DISCUSSION TITLE
Reloading/Updating containers support

### DISCUSSION THREAD

**Discussion by pmioulet** [batect/batect#1158] [Q&A]
Hi again, one thing I miss from docker compose is the ability to update containers that have changed without having to restart the full stack.

When I develop in java I don't usually run the apps in intellij but directly as a containers on my host. So batect simplifies this with the integration of the build and run in one command which is really great. However when I want to test a new change, I'd like to be able to only update the container concerned by the change and not restart my army of containers.

Currently, the only way I see to do this would be to run the dependencies and run the main app I'm working on in separate tasks. Is there a better way of doing this? Is the update of containers something you're planning on looking into? Thanks!

**Comment by charleskorn** [comment]
Hi @pmioulet, thanks for the question. At the moment, reloading containers dynamically isn't on my backlog as it introduces a bunch of interesting other problems (eg. should all other containers that depend on a container that is reloaded also be restarted?)

Does restarting the task take a particularly long time? Or is it just the friction required to manually stop and then restart the task that's annoying?

**Comment by pmioulet** [comment]
Thanks for the quick reply.

It's basically the whole process of stopping and restarting that takes time. Especially given the dependency management based on the health checks is done properly (unlike in docker-compose) which introduces more latency. If there was a way to avoid batect stopping all the containers when the main container is restarted this would also solve the issue as I could manually perform the restart after manually updating it. This could take the form of a daemon functionality maybe?

I understand your point about the dependency management, I kind of suspected this was the reason for the functionality to be missing, I still wanted to make sure I had not missed something!


PS: you might be curious to know that your project has circled round the world as I'm introducing it back into my  Melbournian based customer (I'm myself based in France)

**Reply by charleskorn** [reply]
It's an interesting idea - I'll add it to my (very long) list.

I have no context on your particular situation, so this may not be relevant (or may well be something you've already tried)... it might be worth looking at seeing if anything can be done to improve the startup time of each container and their health checks. For example, if you're starting up a DB server and running a bunch of migrations to prepare the database schema, you could save time by baking a database with all of those schema migrations apply into the image at image build time, rather than running the migrations from scratch each time.

> PS: you might be curious to know that your project has circled round the world as I'm introducing it back into my Melbournian based customer (I'm myself based in France)

Very cool, thanks for letting me know 😃

**Reply by pmioulet** [reply]
Thanks for suggestions, Keycloak end ES just take time to come up, I'll keep using my workaround for now! Good thing about batect is i can keep it in a separate task.
